golang/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+const batchSize = 1000
+
+type Event struct {
+	OrganizationID int                    `json:"organization_id,omitempty"`
+	CustomerID     *int                   `json:"customer_id,omitempty"`
+	CustID         string                 `json:"cust_id,omitempty"`
+	EventName      string                 `json:"event_name,omitempty"`
+	TimeCreated    time.Time              `json:"time_created,omitempty"`
+	Properties     map[string]interface{} `json:"properties,omitempty"`
+	IdempotencyID  string                 `json:"idempotency_id,omitempty"`
+	InsertedAt     time.Time              `json:"inserted_at,omitempty"`
+}
+
+type StreamEvents struct {
+	Events         []Event `json:"events"`
+	OrganizationID int64   `json:"organization_id"`
+	Event          *Event  `json:"event"`
+}
+
+type batch struct {
+	tx         *sql.Tx
+	insertStmt *sql.Stmt
```

Comment:
[nit] but can we name this in full, as in `insertStatement`. What do you think @diego-escobedo though I could make an arguement to name it inline with sql.Stmt